markov-analysis.py

```diff
@@ -0,0 +1,220 @@
+    float x between 0 and 1 might result in something not random, so i think i 
+    will go with the cumulative sum solution from the preview problem
+    another solution might be to just construct a list of each occurrence 
+
+    simple idea:
+    don't use word frequencies, just a set of suffixes
+    so the dict is a mapping P -> S 
+    (p_1, p_2, ..., p_n) -> (s_1, s_2, ..., s_n)
+
+    after finishing:
+    not sure why i ever thought of using a set of suffixes instead of a list,
+    since that would even preserve relative frequencies!
+'''
+
+def mashup(prefix_length, text_length, *titles):
+    '''
+    Generates random text using two or more books as source material.
+
+    TODO: process_file does not properly skip the Project Gutenberg header
+          (see skip_gutenberg_header for more information)
+
+    prefix_length: int, the number of prefix words to use
+    text_length: int, the length of the random text
+    titles: strings, titles of the source books
+    '''
+    word_list = []
+    for title in titles:
+        # WARNING: skip_header only works for emma.txt, Project Gutenberg books
+        #          do NOT have a standard header format
+        word_list = word_list + process_file(title, skip_header=False)
+    prefixes = prefix_map(word_list, prefix_length)
+    return list_to_string(random_text(prefixes, prefix_length, text_length))
+
+
+def generate_text(prefix_length, text_length, title='emma.txt'):
+    '''
+    Generates random text using the specified number of prefixes.
+
+    prefix_length: int, the number of prefix words to use
+    text_length: int, the length of the random text
+    title: string, title of the source book
+
+    int, int -> str
+    '''
+    word_list = process_file(title, skip_header=True)
+    prefixes = prefix_map(word_list, prefix_length)
+    return list_to_string(random_text(prefixes, prefix_length, text_length))
+
+def list_to_string(word_list):
+    '''
+    Converts a list of words to a string, with spaces between each word and a 
+    period at the end.
+
+    word_list: list
+
+    list -> str
+    '''
+    s = ''
+    for word in word_list:
+        s = s + ('%s ' % word)
+    return (s.strip() + '.')
+
+def random_text(prefix_map, prefix_length, text_length):
+    '''
+    Returns a list of words randomly selected by prefix using the given 
+    prefix map.
+
+    # TODO: This looks messy, is there a better way than changing so many types?
+
+    prefix_map: dict of tuples to lists
+    prefix_length: int, the length (in words) of the sequence of prefixes to use
+    m: int, the length (in words) of the list of random words
+
+    dict, int, int -> list
+    '''
+    random_text = list(random.choice(list(prefix_map.keys())))
+
+    while (len(random_text) < text_length):
+        prefix = tuple(random_text[(len(random_text)-prefix_length):len(random_text)])
+        random_text.append(random.choice(prefix_map[prefix]))
+
+    return random_text
```

Comment:
`random_text` isn't really text-- it's a list of strings.  Would prefer the conversion to str happen here.